### PR TITLE
Dummy support of SELECT

### PR DESCRIPTION
**Pull request by tremez**
As some clients require to call SELECT 0, that actually means nothing, lets not crash on it, but simply eat it

### CODE REVIEW

**Comment by doyoubi** [comment]
I think it's a little bit confusing to turn `SELECT` into a no-op. Could you tell us what's the client you're using?

**Comment by tevino** [comment]
He's making a point here.
Yes, Redis Cluster doesn't allow `SELECT` however Corvus is targeting at v2.0 client which do support `SELECT`, I guess allow `SELECT 0` harms nothing.

File: startcorvus.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+killall -9 corvus
+
+/usr/bin/corvus -n 10.254.127.37:6379,10.254.127.48:6379,10.254.127.49:6379 -t 4 -b 6379 -L debug /etc/corvus.conf 1>/var/log/corvus.log 2>/var/log/corvus.err &
```
Comment (tevino) [inline startcorvus.sh]: @tremez Please clean up the PR.

**Comment by tremez** [comment]
Hello
I am a bit confused, as i didnt expect that init/startup scripts would appear in this merge request
However i have removed it
Should i restart pull request, or last commit will appear automatically in the request ?

File: startcorvus.sh (hunk 1 above)
Comment (tremez) [inline startcorvus.sh]: Hello
I am a bit confused, as i didnt expect that init/startup scripts would appear in this merge request
However i have removed it
Should i restart pull request, or last commit will appear automatically in the request ?

**Comment by wooparadog** [comment]
@tremez I believe the correct behavior is to allow `select 0` and return, but when `select` is executed with `db` other than `0`, there should be a clear error stating that it's not supported, to prevent potential data overwriting risks(people think it supports db while it actually doesn't if only seen from an `select` api call).

**Comment by tremez** [comment]
@wooparadog I agree with you, but at this point i wanted to implement only a DUMMY support, to prevent my existing app from crashing due to unsupported SELECT.

**Comment by tevino** [comment]
@tremez I understand your concern, since Corvus is a publicly available project, users may risking their data if we merge implementation like this, limit database to zero should take only a handful lines of code, would you give it a try?

---

You don't need to reopen a PR, using [rebase](https://git-scm.com/docs/git-rebase) is OK.

**Comment by tremez** [comment]
Sorry, i am not a C developer, and as adding support to make SELECT as NOOP was easy, and quite useful - i've suggested it to public.
If anyone would like to fix it - i will appreciate, but i could not do it

**Comment by maralla** [comment]
This is the behaviour of a direct redis server:

redis-select

We should implement according to this.

**Comment by tremez** [comment]
So , will we accept my merge request while "correct" version is not yet done ?

**Comment by wooparadog** [comment]
Sorry, we can't merge this pull request at its current state. However,
there's no preventing you from using your code at your will. But in the
main repo, we appreciate a more complete solution.

**Comment by tevino** [comment]
@tremez This should work in your case: https://github.com/eleme/corvus/pull/115

**Comment by tevino** [comment]
I'm closing this since https://github.com/eleme/corvus/pull/115 was merged, thanks for your work 😄  @tremez

**Pull request closed without merging**